client/hosting-overview/index.tsx

```diff
@@ -23,6 +24,7 @@ export default function () {
 		// @ts-ignore
 		redirectIfCurrentUserCannot( 'manage_options' ),
 		redirectIfP2,
+		redirectIfJetpackNonAtomic,
```

Comment:
Is this actually not related to this PR? I think I am seeing other ongoing PRs modifying the logic around this, so I think we shouldn't include this in this PR, what do you think?

Reply:
Do you have links of those PRs?

Reply:
Sorry, I misremembered; it was https://github.com/Automattic/wp-calypso/pull/90959 but it was about dev tools and it is now closed 😅

Reply:
Wait I'm now a bit confused; in production if I click a site or go to `/overview/` of a fresh new JN site, I get redirected to wp-admin correctly; so what is this change fixing? 🤔

Reply:
hmmm...it's redirected by https://github.com/Automattic/wp-calypso/blob/bafc5210661c1bfee22fb1500661b70d2ab82453/client/sites-dashboard-v2/hooks/use-initialize-dataviews-selected-item.ts#L24 😂

Reply:
I think it's fine to add  `redirectIfJetpackNonAtomic`. The above one doesn't do the redirection at the beginning, so the user will see the site management panel first and get the redirection later. It might be a little weird to users.

Reply:
I see now... the logic is duplicated here:

https://github.com/Automattic/wp-calypso/blob/10aaf653c41e8c6fc3ffb82a4f956190ee1b9cda/client/sites-dashboard-v2/sites-dataviews/dataviews-fields/site-field.tsx#L83-L84

Would be good if we somehow can consolidate these logic! (Not in this PR)

Reply:
> Wait I'm now a bit confused; in production if I click a site or go to /overview/ of a fresh new JN site

well now im confused as well. things change fast 😵 . last i knew we had updated items in the overview for these sites and didn't redirect them.

Given its already redirecting i guess there is no issue adding this here.

Reply:
https://github.com/Automattic/dotcom-forge/issues/7386